WIP: Implement node2vec

cthoyt opened the pull request:
Closes #5

cthoyt committed:
Create node2vec.py

cthoyt committed:
Merge branch 'master' into implement-node2vec

ozlemmuslu commented:
87% ?!? awesome!!!

On Fri, Nov 30, 2018, 17:33 Codecov <notifications@github.com wrote:

> Codecov <https://codecov.io/gh/cthoyt/nrl/pull/6?src=pr&el=h1> Report
>
> ❗️ No coverage uploaded for pull request base (master@7c888a4). Click
> here to learn what that means
> <https://docs.codecov.io/docs/error-reference#section-missing-base-commit>
> .
> The diff coverage is n/a.
>
> [image: Impacted file tree graph]
> <https://codecov.io/gh/cthoyt/nrl/pull/6?src=pr&el=tree>
>
> @@           Coverage Diff            @@
> ##             master      #6   +/-   ##
>
> ========================================
>
>   Coverage          ?   87.6%
>
> ========================================
>
>   Files             ?      12
>
>   Lines             ?     121
>
>   Branches          ?       0
>
> ========================================
>
>   Hits              ?     106
>
>   Misses            ?      15
>
>   Partials          ?       0
>
> ------------------------------
>
> Continue to review full report at Codecov
> <https://codecov.io/gh/cthoyt/nrl/pull/6?src=pr&el=continue>.
>
> *Legend* - Click here to learn more
> <https://docs.codecov.io/docs/codecov-delta>
> Δ = absolute <relative> (impact), ø = not affected, ? = missing data
> Powered by Codecov
> <https://codecov.io/gh/cthoyt/nrl/pull/6?src=pr&el=footer>. Last update
> 7c888a4...43fb56a
> <https://codecov.io/gh/cthoyt/nrl/pull/6?src=pr&el=lastupdated>. Read the comment
> docs <https://docs.codecov.io/docs/pull-request-comments>.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cthoyt/nrl/pull/6#issuecomment-443260488>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEqrsxCNb7NjVTPqUnRjara8W9BT1ZuQks5u0V3mgaJpZM4Y70A->
> .
>

cthoyt committed:
Add testing

cthoyt committed:
update algorithm

ozlemmuslu committed:
Work on node2vec

ozlemmuslu committed:
Add requirements.txt

cthoyt commented on setup.py:
```diff
@@ -5,4 +5,6 @@
-    setuptools.setup()
+    setuptools.setup(name='nrl')
+
+    # TODO: test_requires: node2vec, networkx
```
@ozlemmuslu all that info goes in `setup.cfg` :)

ozlemmuslu committed:
Implement node2vec precompute_probs using igraph

cthoyt committed:
Fix spacing

cthoyt committed:
Fix setup

@ozlemmuslu use `tox -e py` to run tests, or install with `pip install -e .[tests]` to make sure networkx and node2vec are installed

cthoyt committed:
Update node2vec.py

cthoyt committed:
Revert setup

ozlemmuslu committed:
Implement node2vec and add tests

ozlemmuslu committed:
Add abstract random walker class
